opensrp-immunization/src/main/java/org/smartregister/immunization/util/VaccinatorUtils.java

```diff
@@ -871,13 +871,12 @@ public static String getFileName(String fileName, String prefix) {
 
 
     public static List<org.smartregister.immunization.domain.jsonmapping.Vaccine> getVaccineFromVaccineConfigFile(@Nullable android.content.Context context, String vaccines_file) {
-        Map<String, Object> jsonMap = new HashMap<>();
         Class<List<org.smartregister.immunization.domain.jsonmapping.Vaccine>> classType = (Class) List.class;
         Type listType = new TypeToken<List<org.smartregister.immunization.domain.jsonmapping.Vaccine>>() {
         }.getType();
 
         if (specialVaccines == null) {
-            specialVaccines = ImmunizationLibrary.assetJsonToJava(jsonMap, context, vaccines_file, classType, listType);
+            specialVaccines = ImmunizationLibrary.assetJsonToJava(ImmunizationLibrary.getInstance().getVaccinesConfigJsonMap(), context, vaccines_file, classType, listType);
```

Comment:
We probably need to make the specialVaccines variable here local instead of a class filed with a private accesor. It doesn't appear to be used anywhere else.

Reply:
And perhaps rename it from _specialVaccines_ to _vaccines_ since it seems to be invoked by all types

Reply:
It seems it's scope is global for caching purposes. We can probably rename the method name and remove the unused methods calling it. The purpose of the variable seems to stand as holding the specialVaccines only. What do you think?

Reply:
Looks like its only private to this class ?
 _private static List<org.smartregister.immunization.domain.jsonmapping.Vaccine> specialVaccines;_

Reply:
For the _naming and purpose of the variable_, looks like methods like **_getJsonVaccineGroup_** also invoke _getVaccineFromVaccineConfigFile_ and they take only a file name as param, meaning it need not necessarily be a special vaccine file name

Reply:
I guess based its usage here we can leave it as is for now since its used for special vaccines only

Reply:
This code needs to be refactored also to use the main cache since currently there seems to be two caches. [This issue](https://github.com/OpenSRP/opensrp-client-immunization/issues/130) will track and inform how to make the appropriate cleanup changes so that the code is easier to read